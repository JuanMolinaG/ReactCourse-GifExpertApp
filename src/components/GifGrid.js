import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__pulse'>{category}</h3>

      {loading && (
        <p className='animate__animated animate__fadeIn'>Loading...</p>
      )}

      <div className='card-grid'>
        {gifs.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
