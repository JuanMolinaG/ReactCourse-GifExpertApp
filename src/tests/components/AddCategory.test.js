import React from 'react';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Debe de mostrar el componente <AddCategory /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de cambiar el input text', () => {
    const value = 'Hola Mundo';

    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('No debe realizar el POST en submit si input text estÃ¡ vacio', () => {
    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Debe de llamar el setCategories y limpiar el input text', () => {
    const value = 'Hola Mundo';

    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });

    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});
