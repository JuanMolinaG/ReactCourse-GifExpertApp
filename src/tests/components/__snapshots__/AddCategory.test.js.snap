// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrar el componente <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
